From the accuracy values which were obtained in cross validation for decision tree model and the bagging decision tree model. Based on them, the accuracy values shown below are assigned. 
Performing T-Test,where T-values have to be obtained in order to determine the closeness of the models.So to compare them, t-test values have to be compared with the confidence limit. The condition is such that if, t < z or t > -z then, this means that the value of t lies in the rejection region and hence it can be said that the null hypothesis is rejected.  

In [11]:
DT_Accu=[0.71150834, 0.71444863, 0.71330226, 0.7146413,  0.71490815 ,0.71192085,
 0.71440134, 0.71499929, 0.71383468, 0.71420376]

In [12]:
BC_Accu=[0.77448627, 0.77520834, 0.77429135, 0.77413894, 0.77539548, 0.7743107,
 0.77411435, 0.7758724,  0.77550657, 0.77470254]

In [13]:
DT_Selected_Accu=[0.7130373,  0.71437498, 0.71345904, 0.714517,   0.7140706,  0.71300862,
 0.71391235, 0.71436875, 0.71436542, 0.712718  ]

In [14]:
BC_Selected_Accu=[0.77346822, 0.77561176, 0.77382263, 0.77446396, 0.7749403,  0.7749347,
 0.77376141, 0.77526687, 0.77551172, 0.77424841]

In order to calculate the error the formula used is-
error = 1-accuracy

In [15]:
err1=[1-a for a in DT_Accu]
err2=[1-a for a in BC_Accu]
err3=[1-a for a in DT_Selected_Accu]
err4=[1-a for a in BC_Selected_Accu]

In [16]:
print(err1)
print(err2)
print(err3)
print(err4)

[0.28849166000000004, 0.28555136999999997, 0.28669774000000003, 0.28535869999999997, 0.28509185000000004, 0.28807914999999995, 0.28559866, 0.28500071000000005, 0.28616531999999995, 0.28579624000000003]
[0.22551372999999997, 0.22479165999999995, 0.22570864999999996, 0.22586105999999995, 0.22460451999999997, 0.22568929999999998, 0.22588565000000005, 0.22412759999999998, 0.22449343, 0.22529745999999995]
[0.2869627, 0.28562502, 0.28654095999999996, 0.28548300000000004, 0.2859294, 0.28699138, 0.28608765, 0.28563125, 0.28563457999999997, 0.28728200000000004]
[0.22653177999999996, 0.22438824000000002, 0.22617737000000004, 0.22553604000000005, 0.22505969999999997, 0.22506530000000002, 0.22623859000000002, 0.22473312999999995, 0.22448827999999998, 0.22575159]


In [17]:
from scipy import stats

In [18]:
u1=stats.ttest_rel(err1,err2)

In [19]:
u2=stats.ttest_rel(err1,err3)

In [20]:
u3=stats.ttest_rel(err1,err4)

In [21]:
u4=stats.ttest_rel(err2,err3)

In [22]:
u5=stats.ttest_rel(err2,err4)

In [23]:
u6=stats.ttest_rel(err3,err4)

This compares them and gives the required result as shown above and below

In [24]:
print(u1)
print(u2)
print(u3)
print(u4)
print(u5)
print(u6)

Ttest_relResult(statistic=176.09366458070136, pvalue=3.123644977118415e-17)
Ttest_relResult(statistic=-0.11750543642591563, pvalue=0.9090401095235014)
Ttest_relResult(statistic=168.61925720894578, pvalue=4.614705486690977e-17)
Ttest_relResult(statistic=-281.4877498491282, pvalue=4.587050748495944e-19)
Ttest_relResult(statistic=-1.2178739793677618, pvalue=0.25423643666149415)
Ttest_relResult(statistic=285.6000804825084, pvalue=4.0257785283010007e-19)


now, we know that degree of freedom is k-1 and k=10 as nfolds=10
also, z can be calculated as significance level/2 and degree of freedom and we obtain it as z=2.26

Performing null hypothesis for both the models
1st case: u1=>176.0936 >2.26
hence, null hypothesis is rejected. this would mean that the decision tree model is different from the bagging decision model, statistically.

2nd case: u2=> -0.117>-2.26
hence, null hypothesis is not rejected. this would mean that the decision tree model is from the bagging decision model, statistically.

3rd case: u3=> 168.619>2.26
hence, null hypothesis is rejected. this would mean that the decision tree model is different from the bagging decision model, statistically.

4th case: u4=> -281.48<-2.26
hence, null hypothesis is rejected. this would mean that the decision tree model is different from the bagging decision model, statistically.

5th case: u5=> -1.21>-2.26
hence, null hypothesis is not rejected. this would mean that the decision tree model is from the bagging decision model, statistically.

6th case: u6=> 285.6>2.26
hence, null hypothesis is rejected. this would mean that the decision tree model is different from the bagging decision model, statistically.